### PR TITLE
My Bids QA items

### PR DESCRIPTION
**Pull request by erikdstock**
#trivial
The type of this PR is: **enhancement**

### Description
This PR Fixes assorted QA issues with our sale cards and lot list items, documented 🔒  [here](https://www.notion.so/artsy/9-2-20-QA-session-eb29edbf11194c9c97eea3d1ebffb9e4) and specced in figma [here](https://www.figma.com/file/QGuaH3b77El9tQwSMueEHo/Bidding-Management?node-id=1088%3A1776)

#### Header
- [x] Alignment of icon + text
- [x] Correct text for LAI vs non-LAI sales
- [x] Correct text for an in-progress live auction
- [x] Correct line 2 text
#### Lot list items
- [x] Only show "open at"/"ends at" within 10 hours of start time
- [x] Live auctions are never 'Reserve not met'
- [x] Correct text on closed lots 

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))

Example of a narrow display with live sale text wrapping, icon remaining centered with this main text and line below it correctly aligned, purple text while live bidding is in progress:
image
image

Extra timely message when the sale is going live soon:
image

Similar example for a timed auction:
image

Closed lot with correct 'closed on date' text
image

Followup items to ticket
 - sale/lot linking during live bidding
 - cramped text on iphone SE (see active lot photos above)
 - hole in circle star icon shows artwork thumbnail (see last picture above)

[MX-434]: https://artsyproduct.atlassian.net/browse/MX-434

### CODE REVIEW

File: src/lib/Scenes/MyBids/Components/SaleTime.tsx [hunk 1]
```diff
@@ -1,44 +1,69 @@
...
+
+  const noteColor = isLAI ? "purple100" : "black60"
+
+  return (
+    <>
+      <Flex style={{ marginTop: 15 }} flexDirection="row">
+        <Box mr={5} alignSelf="center">
+          <Icon fill={noteColor} />
+        </Box>
+        <Box style={{ flex: -1 }}>
```
Comment (erikdstock) [inline src/lib/Scenes/MyBids/Components/SaleTime.tsx:56]: Worked with @ds300 for a long time trying to debug an issue here - without `flex: -1` the first line overflows its parent container like so:
image

Comment (erikdstock) [reply src/lib/Scenes/MyBids/Components/SaleTime.tsx]: Cleaned up with a clearer presentation of the same issue in [89dcc43](https://github.com/artsy/eigen/pull/3891/commits/89dcc4304762db4186ac19cf04c9982bc3a79510#diff-e3f7217b53d30fb0c33de562c1b0e887R51-R58)

File: src/lib/Scenes/MyBids/helpers/timely.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+export const isLotStandingComplete: (lotStanding: { lotState?: { soldStatus?: string } }) => boolean = (lotStanding) =>
+  !!(lotStanding.lotState?.soldStatus && ["sold", "passed"].includes(lotStanding.lotState.soldStatus?.toLowerCase()))
+
+interface SaleWithTimes {
+  status?: string | null
+  liveStartAt?: string | null
+  endAt?: string | null
+}
+
+export class TimelySale {
```
Comment (erikdstock) [inline src/lib/Scenes/MyBids/helpers/timely.ts:21]: I made one major change to our helpers logic, which was to create a class wrapper/facade for sales that could determine some of our repetitive calculations that depend on presence of dates, etc.

File: src/lib/Scenes/MyBids/index.tsx [hunk 3]
```diff
@@ -31,9 +31,11 @@ class MyBids extends React.Component<MyBidsProps> {
 
     const [activeStandings, closedStandings] = partition(
       lotStandings.filter((ls) => !!ls),
-      (ls) => lotInActiveSale(ls)
+      (ls) => !TimelySale.create(ls?.saleArtwork?.sale!).isClosed
     )
 
+    // const saleIds = sortSaleIds(activeStandings, watchedLots)sa
```
Comment (erikdstock) [inline src/lib/Scenes/MyBids/index.tsx:37]: will remove this pseudocode

File: src/lib/Scenes/MyProfile/MyProfile.tsx [hunk 4]
```diff
@@ -28,7 +27,6 @@ const MyProfile: React.FC<{ me: MyProfile_me; relay: RelayRefetchProp }> = ({ me
       listRef.current?.scrollToOffset({ offset: 0, animated: false })
     })
   }, [])
-  const activeBidCount = extractNodes(me.auctionsLotStandingConnection).filter((ls) => !lotInActiveSale(ls)).length
```
Comment (erikdstock) [inline src/lib/Scenes/MyProfile/MyProfile.tsx:31]: this stuff broke as part of other changes and is no longer part of design, so i've left only the menu item as an entry point for development for now.

File: src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    describe("live bidding in progress", () => {
+      let tree: ReactTestRenderer
+      beforeEach(() => {
+        const liveStartAt = moment().subtract(5, "minutes").toJSON()
+        tree = renderWithWrappers(<SaleInfo sale={saleFixture({ liveStartAt, status: "open" })} />)
+      })
+      it("has a purple100 icon + note if live bidding is active, black60 otherwise", () => {
+        expect(tree.root.findByType(BoltFill).props.fill).toMatch("purple100")
+      })
+      it("has purple100 text text", () => {
```
Comment (erikdstock) [inline src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx:57]: lots of new tests here.

File: src/lib/Scenes/MyBids/__fixtures__/MyBidsQuery.ts [hunk 6]
```diff
@@ -17,21 +16,20 @@ const swannSale = {
...
-const heritageSale = {
+// an open, LAI auction
+const laiSale = {
   internalID: "heritage",
   saleType: "auction",
   href: "/auction/heritage-urban-art-summer-skate",
-  endAt: "2020-08-05T15:00:00+00:00",
+  liveStartAt: "2020-08-05T15:00:00+00:00",
   status: "open",
-  displayTimelyAt: "live in 2d",
```
Comment (erikdstock) [inline src/lib/Scenes/MyBids/__fixtures__/MyBidsQuery.ts:34]: we are no longer relying on this string from metaphysics.

File: src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx (hunk 5 above)
Comment (bhoggard) [reply src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx:57]: "text text" ?

File: src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+
+        const lines = tree.root.findAllByType(Text)
+        expect(lines.length).toEqual(2)
+        expect(lines[1].props.color).toEqual("yellow100")
+
+        expect(text).toContain("Live bidding begins today at 2:00am")
+        expect(text).toContain("Opens in 38 minutes")
+      })
+
+      it("displays 'Live bidding begins today' and time if the sale closes on the same calendar day", () => {
```
Comment (bhoggard) [inline src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx:91]: Using "closes" when setting `liveStartAt` is probably going to be confusing to anyone not very familiar with LAI sales.

**Comment by bhoggard** [comment]
LGTM. I had a few minor test phrasing questions.

**Review by bhoggard — APPROVED** [review]

File: src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx (hunk 5 above)
Comment (erikdstock) [reply src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx:57]: eactly 👌

File: src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx (hunk 7 above)
Comment (erikdstock) [reply src/lib/Scenes/MyBids/__tests__/SaleInfo-tests.tsx:91]: oh, thanks, copied test string.

**Pull request merged**